designs/generic-git-provider-gitops.md

```diff
@@ -0,0 +1,205 @@
+      username: danbudris
+      privateKeyFile: '/path/to/my/private_key.rsa'
+      branch: branchToUseInRemoteRepo
+```
+
+The CLI will then validate if the repository exists; if it does not exist, it will return a preflight validation error, as we cannot create a repository in a remote with basic auth credentials and no access to the API.
+
+If it does exist, the installation will proceed in the same manner as the current Github integration, cloning the repository locally, initializing it if needed, and committing the EKS Anywhere/Flux system configuration files and pushing the contents to the remote.
+
+### Delete
+The delete experience will bypass the deletion of the repository if the generic git provider is used.
+
+## Solution Details
+EKS Anywhere uses the `flux bootstrap` command to bootstrap the cluster with the Flux controllers and configure them to use the given provider. We translate the user-provided configuration in the GitOpsConfig spec into command line arguments for the Flux executable and, in the case of a specific provider like GitHub, to calls to the Github API to take specific actions like creating repositories. 
+
+Up until now, we've only called [`flux bootstrap github`](https://fluxcd.io/docs/cmd/flux_bootstrap_github/) and passed in the Github specific configuration along with the Flux specific configuration. However, Flux has a generic git bootstrap command, [`flux bootstrap git`](https://fluxcd.io/docs/cmd/flux_bootstrap_git/), which takes generic basic authentication parameters such as SSH Private key path, password, and username. 
+
+We will take advantage of this fact to translate a basic, provider-agnostic `git` provider in our GitOpsConfiguration spec to arguments for `flux bootstrap git`, bypassing the provider-specific steps along the way.
+
+So, how does will this new sequence flow? Let's take a look at how the GitOps bootstrap process would flow during cluster creation.
+
+![GitOps Bootstrap Flow](images/gitops-flow.png)
+
+-  check if the user provided repository exists, using a git provider specific implementation
+    - generic git provider will use `git ls-remote` or similar to validate existence of remote and branch ref. 
+    - Provider specific methods will be used for specific providers like GitHub, using the provider API to check the status of the remote.
+- if the repository does not exist, we determine if we can create it for the user; if we can, we will do so, and initialize it. 
+- if the repository does not exist, we will proceed as we do today, cloning and initializing it as needed.
+
+Preflight validations will be executed to check the existence of the repository at CLI run time; if it does not exist, and the provider does not support repository creation, we will return a validation error. We will perform another validation durning the flux installation tasks to take care of cases such as the repository being deleted between preflight validations and the start of the task.
```

Comment:
Probably no need to perform another validation, the push will fail and that is the only indicator that maters